### ISSUE TITLE
Node view Page

### ISSUE THREAD

**Issue by laylaylo** [bounswe/bounswe2023group9#321]
**Issue Description**
A page should be designed and implemented to display details of a node including content, contributor info and question/answers.

**Steps**
- [x] Design
- [x] Implementation 
- [x] Navigation to Other Screens
- [x]  Screen Tests
 
**Deadline**
31.10.2023

**Comment by Zulalm** [comment]
A structure for the node view page is designed and implemented. Hardcoded data will be included in the code until a provider for the page is implemented. 
Here is the view of the page. I can improve the design based on the feedbacks.
![image](https://github.com/bounswe/bounswe2023group9/assets/111250696/77f0804f-c53e-416a-8fb3-2e1e740af00a)

**Comment by Zulalm** [comment]
A provider to get node details is implemented. Since there is not any API call to get references and citations hardcoded date is left to visualize. Questions and answers are not implemented on the backend yet, but I created some mock data to show how the content might be displayed in the questions tab. Design of the tabs might be revised based on the feedbacks.

**Comment by Zulalm** [comment]
I could not implement the navigation functionality by myself. I requested help from @laylaylo  and @defabdullah. We added onTap functionality to references, citations and contributors of a theorem that navigates application to the detail page of reference, citation or profile page of the contributor. Screen tests are completed collaboratively with team members and I fixed the bugs that are found.  The branchs that I developed the code are merged into frontend.
All requirements related with node view page we intended to provide for this milestone is completed. Therefore, I am closing this issue. 
 
**Remaining Issues Related to Node View Page**
* Questions and answers page should be connected to the backend. It's design might be changed.
* Fetching data from the backend should be improved. It takes too long to load the page.
